### PR TITLE
track queue time via queued_at field from payload

### PR DESCRIPTION
**Pull request by igorwwwwwwwwwwwwwwwwwwww**
This patch will give us a metric of the actual queue wait time experienced by users.

Depends on https://github.com/travis-ci/travis-scheduler/pull/55

### CODE REVIEW

**Comment by henrikhodne** [comment]
Note that `queued_at` is reset in a requeue, so this won't reflect longer times when requeues are happening, so the reported numbers will probably be too low. Probably better than nothing, though.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Good point, that is something I'd want to track separately though. I think with queue wait + requeue we already have a pretty good idea of how long things are taking.

It's also worth noting that because we overwrite that information it is currently very difficult to even get that kind of number. AFAIK the only way at the moment is to look through log files.

**Comment by henrikhodne** [comment]
Would it be a good idea to split up the metric per queue so we can tell what the queue time for each individual queue is?

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
@henrikhodne imo all metrics should be split per queue -- if that is not the case yet, we should change that. wdyt?

**Comment by henrikhodne** [comment]
> imo all metrics should be split per queue -- if that is not the case yet, we should change that. wdyt?

Come to think of it they are split per worker hostname in the source, so we can do some globbing to make charts that are split per worker name and do it that way. So ignore what I said, I forgot we already do this :P

File: job_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"ssh_key": null,
+	"timeouts": {
+		"hard_limit": null,
+		"log_silence": null
+	},
+	"cache_settings": {}
+}
+`
+
+func TestUnarshalJobPayload(t *testing.T) {
```
Comment (meatballhat) [inline job_test.go]: `s/Unarshal/Unmarshal/`

Comment (meatballhat) [inline job_test.go]: uh whoops I meant to hit "Request Changes" :woman_shrugging:

**Pull request merged**